### PR TITLE
[AD-176] Update wallet-new from iohk

### PR DESCRIPTION
**Pull request by maksbotan**

### CODE REVIEW

File: ariadne/src/Ariadne/Wallet/Cardano/Kernel/DB/HdWallet/Read.hs [hunk 1]
```diff
@@ -39,6 +40,8 @@ import Ariadne.Wallet.Cardano.Kernel.DB.Spec
 import Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet (IxSet)
 import qualified Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet as IxSet
 
+{-# ANN module ("HLint: ignore Unnecessary hiding" :: Text) #-}
```
Comment (maksbotan) [inline ariadne/src/Ariadne/Wallet/Cardano/Kernel/DB/HdWallet/Read.hs]: Not needed now?

Comment (Pastafarianist) [reply ariadne/src/Ariadne/Wallet/Cardano/Kernel/DB/HdWallet/Read.hs]: Right, forgot about that one. Fixed.

**Comment by Pastafarianist** [comment]
Remaining issues:

- [ ] Update address checkpoints in `applyBlock` in `Ariadne.Wallet.Cardano.Kernel.DB.AcidState`
- [ ] Get rid of `error ` in `createPrefiltered` in `Ariadne.Wallet.Cardano.Kernel.DB.AcidState`
- [ ] Make sure there are no other places where address checkpoints have to be updated

**Comment by Pastafarianist** [comment]
@gromakovsky please review.

I understand that this looks like a very big change, so I made [this diff](https://gist.github.com/Pastafarianist/37375fc9b644af0e45e59eba91077a03) for you. This is a diff between a raw import and this PR. It was made in the following way:

1. I followed the instructions in [`updating_wallet.md`](https://github.com/serokell/ariadne/blob/master/ariadne/updating_wallet.md) through to step 6, inclusive.
2. `git rebase master -s recursive -X theirs` (my `master` is at `92592646e7c987cd69e0e71d6e7225413a4376de`)
3. `git diff HEAD..maksbotan/ad-176-update-15-06-2018 > upstream.patch`

File: ariadne/src/Ariadne/Wallet/Cardano/Kernel/DB/HdWallet.hs [hunk 2]
```diff
@@ -368,3 +387,87 @@ zoomHdAddressId embedErr addrId =
...
+    build (HdAccountIx ix)
+        = bprint ("HdAccountIx: "%build) ix
+
+instance Buildable HdAccountId where
+    build (HdAccountId parentId accountIx)
+        = bprint ("HdAccountId: "%build%", "%build) parentId accountIx
+
+instance Buildable HdAddressChain where
+    build HdChainInternal = bprint "internal"
+    build HdChainExternal = bprint "external"
```
Comment (gromakovsky) [inline ariadne/src/Ariadne/Wallet/Cardano/Kernel/DB/HdWallet.hs]: `bprint`s are redundant.

**Comment by Pastafarianist** [comment]
@gromakovsky it does work as expected, I tested that part separately.

**Pull request merged**